File: frontend/test-action-api.py
#!/usr/bin/env python3
"""
Test script to send current action updates to the Cluely widget
Run this with: python3 test-action-api.py

This demonstrates how your Flask server can send updates to the widget
"""

import requests
import time

def send_current_action(action):
    """Send current action to Cluely widget"""
    try:
        response = requests.post(
            'http://localhost:3456/api/currentaction',
            data=action,
            headers={'Content-Type': 'text/plain'}
        )
        print(f"Sent: {action}")
        print(f"Response: {response.status_code} - {response.json()}\n")
        return response.json()
    except Exception as e:
        print(f"Error sending action: {e}\n")
        return None

if __name__ == "__main__":
    print("Sending test actions to Cluely widget...\n")

    # Simulate agent actions
    send_current_action("Agent is thinking...")
    time.sleep(2)

    send_current_action("Opening calendar app...")
    time.sleep(2)

    send_current_action("Creating new event...")
    time.sleep(2)

    send_current_action("Setting event time for tomorrow at 12pm...")
    time.sleep(2)

    send_current_action("Event created successfully!")

    print("\nAll actions sent!")

File: frontend/test-action-api.js
// Test script to send current action to the Cluely widget
// Run this with: node test-action-api.js

const http = require('http');

function sendCurrentAction(action) {
  const data = action;

  const options = {
    hostname: 'localhost',
    port: 3456,
    path: '/api/currentaction',
    method: 'POST',
    headers: {
      'Content-Type': 'text/plain',
      'Content-Length': Buffer.byteLength(data)
    }
  };

  const req = http.request(options, (res) => {
    console.log(`Status: ${res.statusCode}`);

    res.on('data', (chunk) => {
      console.log(`Response: ${chunk}`);
    });
  });

  req.on('error', (error) => {
    console.error(`Error: ${error.message}`);
  });

  req.write(data);
  req.end();
}

// Test with different actions
console.log('Sending test actions to Cluely widget...\n');

sendCurrentAction('Opening calendar app...');

setTimeout(() => {
  sendCurrentAction('Creating new event...');
}, 2000);

setTimeout(() => {
  sendCurrentAction('Setting event time for tomorrow at 12pm...');
}, 4000);

setTimeout(() => {
  sendCurrentAction('Event created successfully!');
}, 6000);
